subsys/bluetooth/controller/ll_sw/ctrl.c

```diff
@@ -1551,7 +1560,8 @@ static inline u32_t isr_rx_scan(u8_t devmatch_ok, u8_t devmatch_id,
 		  ((pdu_adv_rx->type == PDU_ADV_TYPE_EXT_IND) &&
 		   (_radio.scanner.phy)) ||
 #endif /* CONFIG_BLUETOOTH_CONTROLLER_ADV_EXT */
-		  ((pdu_adv_rx->type == PDU_ADV_TYPE_SCAN_RSP) &&
+		  ((pdu_adv_rx->type == PDU_ADV_TYPE_SCAN_RSP &&
+		    isr_scan_rsp_adva_check(pdu_adv_rx)) &&
```

Comment:
move this after state != 0, so that it short-circuits early (under cases where scanner rx-es scan_rsp under passive scanning, for example).